plugins/importexport/native/filter/NativeXmlIssueGalleyFilter.inc.php

```diff
@@ -84,33 +84,45 @@ function handleElement($node) {
 				break;
 			case 'label': $issueGalley->setLabel($n->textContent); break;
 			case 'issue_file':
-				$issueFileDao = DAORegistry::getDAO('IssueFileDAO'); /* @var $issueFileDao IssueFileDAO */
-				$issueFile = $issueFileDao->newDataObject();
-				$issueFile->setIssueId($issue->getId());
-
-				for ($o = $n->firstChild; $o !== null; $o=$o->nextSibling) if (is_a($o, 'DOMElement')) switch($o->tagName) {
-					case 'file_name': $issueFile->setServerFileName($o->textContent); break;
-					case 'file_type': $issueFile->setFileType($o->textContent); break;
-					case 'file_size': $issueFile->setFileSize($o->textContent); break;
-					case 'content_type': $issueFile->setContentType((int)$o->textContent); break;
-					case 'original_file_name': $issueFile->setOriginalFileName($o->textContent); break;
-					case 'date_uploaded': $issueFile->setDateUploaded($o->textContent); break;
-					case 'date_modified': $issueFile->setDateModified($o->textContent); break;
-					case 'embed':
-						import('classes.file.IssueFileManager');
-						$issueFileManager = new IssueFileManager($issue->getId());
-						$filePath = $issueFileManager->getFilesDir() . $issueFileManager->contentTypeToPath($issueFile->getContentType()) . '/' . $issueFile->getServerFileName();
-						$issueFileManager->writeFile($filePath, base64_decode($o->textContent));
-						break;
-				}
-
-				if (!$issueGalley->getFileId()) {
+				try	{
+					$issueFileDao = DAORegistry::getDAO('IssueFileDAO'); /* @var $issueFileDao IssueFileDAO */
+					$issueFile = $issueFileDao->newDataObject();
+					$issueFile->setIssueId($issue->getId());
+
+					for ($o = $n->firstChild; $o !== null; $o=$o->nextSibling) {
+						if ($o instanceof DOMElement) {
+							switch($o->tagName) {
+								case 'file_name': $issueFile->setServerFileName($o->textContent); break;
+								case 'file_type': $issueFile->setFileType($o->textContent); break;
+								case 'file_size': $issueFile->setFileSize($o->textContent); break;
+								case 'content_type': $issueFile->setContentType((int)$o->textContent); break;
+								case 'original_file_name': $issueFile->setOriginalFileName($o->textContent); break;
+								case 'date_uploaded': $issueFile->setDateUploaded($o->textContent); break;
+								case 'date_modified': $issueFile->setDateModified($o->textContent); break;
+								case 'embed':
```

Comment:
Does the schema requires a specific ordering of elements?
If not, and if the `<embed>` is places before the other elements, the insert will be attempted without having all information available.